.github/workflows/main.yml

```diff
@@ -46,7 +46,7 @@ jobs:
         uses: actions/checkout@v2
 
       - name: go cache
-        uses: actions/cache@v1
+        uses: actions/cache@v4
```

Comment:
nit: could use cache@v5 too.

Reply:
Hmm cache/v5 doesn't seem to exist. Reverting to v4.

Reply:
Actually, we need to use `setup-go@v5` which then has caching enabled automatically, then the cache step can be removed.

Reply:
Sorry, meant `setup-go@v5` just somehow wrote cache.